### PR TITLE
Change dependency on curve25519-dalek-ng to curve25519-dalek

### PR DESCRIPTION
**Pull request by Stentonian**
Closes https://github.com/zkcrypto/bulletproofs/issues/15

@cathieyun 

The major version has to be bumped because importing both Bulletproofs & curve25519-dalek-ng after this change will result in conflicts.

I ran both the tests and benches:
```bash
cargo test
cargo bench
```

I also tested that importing Bulletproofs 5.0.0 & curve25519-dalek 4.1.1 compiles successfully (some changes may need to be done, depending on how curve25519-dalek is used.

### CODE REVIEW

**Comment by cathieyun** [comment]
LGTM, thanks for putting this PR together.

**Review by cathieyun — APPROVED** [review]

File: src/range_proof/messages.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use alloc::vec::Vec;
 use core::iter;
 use curve25519_dalek::ristretto::{CompressedRistretto, RistrettoPoint};
 use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::VartimeMultiscalarMul;
```
Comment (cathieyun) [inline src/range_proof/messages.rs:13]: why is this additional trait import needed?

Comment (Stentonian) [reply src/range_proof/messages.rs:13]: I actually just moved this value here from line 94 (and removed 1 unused trait). I can move it back again to line 94 if you prefer? This trait is needed to expose the following function on line 128:
```rust
let P_check = RistrettoPoint::vartime_multiscalar_mul(
```

**Comment by Stentonian** [comment]
Thanks for reviewing @cathieyun !

File: src/range_proof/messages.rs (hunk 1 above)
Comment (cathieyun) [reply src/range_proof/messages.rs:13]: That seems fine, thanks for explaining. LGTM

**Pull request merged**